File: lec02/lec02q5.cpp
// Rotation of a number by r 

#include <iostream>
using namespace std;
int main(){

    int n;
    cin>> n;
    int dn =n;
    int r;
    cin>> r;
    int dr= r;

    int length=0;
    while(dn!=0){
        length++;
        dn=dn/10;
    }

    r=r%length;

    if(r<0){
        r= length+r;
    }
    int digits=length-r;
    
    int rem=0;
    int ans=0;
    int pow=1;

    while(r>0){
        int rem= n%10;
        n=n/10;
        ans= ans + pow*rem;
        pow=pow*10;
        r--;
    }

    
    while(digits>0){
        ans = ans*10;
        digits--;
    }

    ans=ans +n;
    cout << ans;




    return 0;
}

File: lec02/lec02q5.java
import java.util.*;
class lec02q5{
    public static void main(String[] args){
        Scanner scn= new Scanner(System.in);
        
        int n= scn.nextInt();
        int copy =n;
        int r= scn.nextInt();
        int ans=0;
        int rem=0;
        int suf=0;
        
        int digits=0;
        while(copy>0){
            digits++ ;
            copy=copy/10;
        }
        int mul=digits-r;

        int power =1;
        // for(int i=0; i<digits; i++){
        //     power= power*10;
        // }
        
        while(r>0){
            rem = n%10;
            n=n/10;
            ans= ans + rem*power;
            power*=10;
            r--;
           
        }
       

        for(int i=1; i<=(mul); i++){
                ans= ans*10;
             }
        
        ans= ans+n;
        System.out.println(ans);

        
    }
}